src/main/java/org/dependencytrack/event/kafka/componentmeta/IntegrityCheck.java

```diff
@@ -0,0 +1,92 @@
+package org.dependencytrack.event.kafka.componentmeta;
+
+import alpine.Config;
+import alpine.common.logging.Logger;
+import org.datanucleus.util.StringUtils;
+import org.dependencytrack.common.ConfigKey;
+import org.dependencytrack.model.Component;
+import org.dependencytrack.model.IntegrityAnalysis;
+import org.dependencytrack.model.IntegrityMatchStatus;
+import org.dependencytrack.model.IntegrityMetaComponent;
+import org.dependencytrack.persistence.QueryManager;
+import org.hyades.proto.repometaanalysis.v1.AnalysisResult;
+
+import java.util.Date;
+
+import static org.dependencytrack.model.IntegrityMatchStatus.COMPONENT_MISSING_HASH;
+import static org.dependencytrack.model.IntegrityMatchStatus.COMPONENT_MISSING_HASH_AND_MATCH_UNKNOWN;
+import static org.dependencytrack.model.IntegrityMatchStatus.HASH_MATCH_FAILED;
+import static org.dependencytrack.model.IntegrityMatchStatus.HASH_MATCH_PASSED;
+import static org.dependencytrack.model.IntegrityMatchStatus.HASH_MATCH_UNKNOWN;
+
+public class IntegrityCheck {
+
+    private static final Logger LOGGER = Logger.getLogger(IntegrityCheck.class);
+
+    public static void performIntegrityCheck(final IntegrityMetaComponent integrityMetaComponent, final AnalysisResult result, final QueryManager qm) {
+        if (!Config.getInstance().getPropertyAsBoolean(ConfigKey.INTEGRITY_CHECK_ENABLED)) {
+            LOGGER.debug("Integrity check is disabled");
+            return;
+        }
+        if (StringUtils.isEmpty(result.getComponent().getUuid())) {
+            LOGGER.info("Result received on topic does not have component uuid, integrity check cannot be performed");
+            return;
+        }
+        //if integritymetacomponent is  null, try to get it from db
+        //it could be that integrity metadata is already in db
+        IntegrityMetaComponent metadata = integrityMetaComponent == null ? qm.getIntegrityMetaComponent(result.getComponent().getPurl().toString()) : integrityMetaComponent;
```

Comment:
we can move metadata query after we check existence of component in db just below this.

Reply:
Good spot, thanks